spring-cloud-stream-metrics/src/main/java/org/springframework/cloud/stream/metrics/BinderMetricsEmitter.java

```diff
@@ -122,9 +122,11 @@ public void onApplicationEvent(ContextRefreshedEvent event) {
 				if (source instanceof EnumerablePropertySource) {
 					EnumerablePropertySource e = (EnumerablePropertySource) source;
 					for (String propertyName : e.getPropertyNames()) {
-						for (String relaxedPropertyName : new RelaxedNames(propertyName)) {
+						RelaxedNames relaxedNames = new RelaxedNames(propertyName);
+						outer: for (String relaxedPropertyName : relaxedNames) {
```

Comment:
the label "outer" is a bit confusing here since this for loop is actually itself nested in another for loop